server/mqtt.go

```diff
@@ -2691,8 +2675,14 @@ func (sess *mqttSession) clear() error {
 	}
 
 	if seq > 0 {
-		if err := sess.jsa.deleteMsg(mqttSessStreamName, seq, true); err != nil {
-			return fmt.Errorf("unable to delete session %q record at sequence %v", id, seq)
+		err := sess.jsa.deleteMsg(mqttSessStreamName, seq, true)
+		// Ignore the various errors indicating that the message (or sequence)
+		// is already deleted, can happen in a cluster.
+		if isErrorOtherThan(err, JSSequenceNotFoundErrF) {
+			if !IsNatsErr(err, JSStreamMsgDeleteFailedF) ||
+				!strings.HasSuffix(err.Error(), "no message found (10057)") {
```

Comment:
Not sure about that one, it feels a bit brittle. You exclude errors related to sequence not found (`isErrorOtherThan(err, JSSequenceNotFoundErrF)`), so here, I think what you are trying to exclude is the specific error about message being not found. The server would return a `JSStreamMsgDeleteFailedF` error, which is more generic, but you want to ignore only if the "sub-reason" is message not found, correct? If so, you should test only for the `JSNoMessageFoundErr` error, and I believe it could be done in one check:
```
if isErrorOtherThan(err, JSSequenceNotFoundErrF) && isErrorOtherThan(err, JSNoMessageFoundErr) {
    return fmt.Errorf(...
}
```

Reply:
Unfortunately, the underlying error was an `errors.stringError` with the formatted string. I did not chase it deep enough to see where it got converted, but I agree that ^^ is too fragile. Will try to see if I can fix it.

Reply:
"kinda" fixed now, please LMK if there already is a function to map those errors. @kozlovic please see changes in stream.go